docs/tutorial/write-your-first-kubernetes-charm-for-a-django-app.rst

```diff
@@ -1,699 +1,781 @@
+.. literalinclude:: code/django/task.yaml
+    :language: bash
+    :start-after: [docs:charm-init]
+    :end-before: [docs:charm-init-end]
+    :dedent: 2
 
-    charmcraft init --profile django-framework --name django-hello-world
+The files will automatically be created in your working directory.
 
-The files will automatically be created in your working directory. We will need to
-connect to the PostgreSQL database. Open the project file and add the following section
-to the end of the file:
+We will need to connect the Django application to the PostgreSQL database.
+Open the ``charmcraft.yaml`` file and add the following section to the end
+of the file:
 
-.. code-block:: yaml
+.. literalinclude:: code/django/postgres_requires_charmcraft.yaml
+   :language: yaml
 
-    requires:
-      postgresql:
-        interface: postgresql_client
-        optional: false
-        limit: 1
+Now let’s pack the charm:
 
-The charm depends on several libraries. Download the libraries and pack
-the charm:
-
-.. code-block:: bash
-
-    CHARMCRAFT_ENABLE_EXPERIMENTAL_EXTENSIONS=true charmcraft fetch-libs
-    CHARMCRAFT_ENABLE_EXPERIMENTAL_EXTENSIONS=true charmcraft pack
+.. literalinclude:: code/django/task.yaml
+    :language: bash
+    :start-after: [docs:charm-pack]
+    :end-before: [docs:charm-pack-end]
+    :dedent: 2
 
 .. note::
 
-    Depending on your network, this step can take a couple of minutes
-    to finish.
+   ``CHARMCRAFT_ENABLE_EXPERIMENTAL_EXTENSIONS=true`` may be required
+   in the pack command for older versions of Charmcraft.
 
-Once Charmcraft has finished packing the charm, you'll find a new file in
-your working directory with the charm extension. View its contents:
+Depending on your system and network, this step can take several
+minutes to finish.
 
-.. code-block:: bash
-
-    ls *.charm -l
+Once Charmcraft has finished packing the charm, the terminal will
+respond with something similar to
+``Packed django-hello-world_ubuntu-22.04-amd64.charm``.
 
 .. note::
 
-    If you changed the project name or are not on the ``amd64`` platform, the name of
-    the ``.charm`` file will be different for you.
-
+   If you are not on the ``amd64`` platform, the name of the ``.charm``
+   file will be different for you.
 
 Deploy the Django application
 -----------------------------
 
-A Juju model is needed to deploy the application. Create a new model:
+A Juju model is needed to handle Kubernetes resources while deploying
+the Django application. Let’s create a new model:
 
-.. code-block:: bash
+.. literalinclude:: code/django/task.yaml
+    :language: bash
+    :start-after: [docs:add-juju-model]
+    :end-before: [docs:add-juju-model-end]
+    :dedent: 2
 
-    juju add-model django-hello-world
+If you are not on a host with the ``amd64`` architecture, you will need
+to include a constraint to the Juju model to specify your architecture.
+You can check the architecture of your system using
+``dpkg --print-architecture``.
```

Comment:
I find this odd, though I don't have a machine to test against. Since the model is backed by our local microk8s, why wouldn't its architecture constraint already be for that architecture?

Reply:
I've asked in the Juju matrix channel - it seems like a strange behaviour by Juju that could probably use its own infobox.

Reply:
OK, the answer is that Juju always defaults to amd64... Interesting.